registry/storage/registry.go

```diff
@@ -12,47 +12,82 @@ import (
 // package. All instances should descend from this object.
 type registry struct {
 	blobStore                   *blobStore
-	blobServer                  distribution.BlobServer
-	statter                     distribution.BlobStatter // global statter service.
+	blobServer                  *blobServer
+	statter                     *blobStatter // global statter service.
 	blobDescriptorCacheProvider cache.BlobDescriptorCacheProvider
 	deleteEnabled               bool
 	resumableDigestEnabled      bool
 }
 
+// RegistryOption is the type used for functional options for
+// NewRegistryWithDriver.
+type RegistryOption func(*registry)
```

Comment:
I know we aren't returning errors for `NewRegistryWithDriver` but these should probably return an error if they cannot be applied.